PowerEditor/src/WinControls/TaskList/TaskList.cpp

```diff
@@ -105,33 +105,32 @@ RECT TaskList::adjustSize()
 	RECT rc;
 	ListView_GetItemRect(_hSelf, 0, &rc, LVIR_ICON);
 	const int imgWidth = rc.right - rc.left;
-	const int leftMarge = 30;
-	const int xpBottomMarge = 5;
-	const int w7BottomMarge = 15;
+	const int aSpaceWidth = ListView_GetStringWidth(_hSelf, TEXT(" "));
+	const int leftMarge = ::GetSystemMetrics(SM_CXFRAME) * 2 + aSpaceWidth * 4;
 
 	// Temporary set "selected" font to get the worst case widths
 	::SendMessage(_hSelf, WM_SETFONT, reinterpret_cast<WPARAM>(_hFontSelected), 0);
 	int maxwidth = -1;
 
-	_rc.left = 0;
-	_rc.top = 0;
-	_rc.bottom = 0;
+	_rc = { 0, 0, 0, 0 };
+	TCHAR * buf = new TCHAR[MAX_PATH];

```

Comment:
Because potentially allocating 32k on the stack is probably not a good idea. MS tries to fix their scrap in their API and the real limit of a path is 32k. So this value may change in the future and there is no good reason to use a static buffer for a path.